### PR TITLE
Event dispatching

### LINKED ISSUE #126: Add Events
- [ ] preRemove
 - [ ] prePersist
 - [ ] preUpdate
 - [ ] prePost
 - [ ] preFlush

### PR DESCRIPTION
**Pull request by einorler**
Rebased the PR from tompoc and modified the events to be more compatible with the bundle logic.

Closes #126

### CODE REVIEW

**Comment by einorler** [comment]
Blocks #660

File: Event/BulkEvent.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ONGR\ElasticsearchBundle\Event;
+
+use Symfony\Component\EventDispatcher\Event;
+
+class BulkEvent extends Event
```
Comment (saimaz) [inline Event/BulkEvent.php]: Add also setters for each variable.

File: Event/CommitEvent.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ONGR\ElasticsearchBundle\Event;
+
+use Symfony\Component\EventDispatcher\Event;
+
+class CommitEvent extends Event
```
Comment (saimaz) [inline Event/CommitEvent.php]: Add also setters for each variable.

File: Event/Events.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+/**
+ * Contains all events thrown in the ONGRElasticsearchBundle
+ */
+final class Events
+{
+    /**
+     * The BULK event occurs before during the processing of bulk method
+     */
+    const BULK = 'es.pre_index';
```
Comment (saimaz) [inline Event/Events.php]: name it `es.bulk`

File: Service/Manager.php [hunk 4]
```diff
@@ -265,6 +283,8 @@ public function persist($document)
         $documentArray = $this->converter->convertToArray($document);
         $type = $this->getMetadataCollector()->getDocumentType(get_class($document));
 
+        $this->eventDispatcher->dispatch(Events::PERSIST, new PersistEvent($document));
```
Comment (saimaz) [inline Service/Manager.php]: I think this even is not necessary. To modify all documents we can in pre_commit event.

**Pull request merged**